### ISSUE TITLE
support non-standard characters in symbol names on command line

### ISSUE THREAD

**Issue by ahh** [janestreet/magic-trace#69]
I have a binary with a symbol whose c name according to e.g. readelf is like:

```
foo_bar_$5_baz$3_qux_12345
```

This fails:
```
magic-trace attach <stuff> -symbol 'foo_bar_$5_baz$3_qux_12345'
```
Surprisingly if I do:
```
magic-trace attach <stuff> -symbol foo_bar
```
I see the right symbol in my fzf, can select it, and everything works (it does not appear to have any escapes etc.)  I wonder if we're not escaping the characters in a regex? I also wonder if we should separate symbol-regex from this-is-the-symbol-i-know-it.

**Comment by ahh** [comment]
(unsurprisingly, this is trivial to replicate with an ocaml anonymous function name.)

**Comment by cgaebel** [comment]
How do you all feel about only supporting *exact* symbol name matches? I worry about cases like two symbols existing `symbol` and `symbol_2` and getting traces triggered off of `symbol_2` when I really want to only trace symbol.

I suppose we could offer multiple ways to select symbols, but that adds more complexity to our UI.

**Comment by cgaebel** [comment]
Fixed by #79.

**Comment by ahh** [comment]
It's an interesting point; I think the fzf is a really useful, dare I say
magic feature but confusion can happen.

We could special case exact match?
